src/enzo/ActiveParticle_SmartStar.C

```diff
@@ -0,0 +1,1058 @@
+      static_cast<ActiveParticleType_SmartStar*>(
+              thisGrid->ActiveParticles[n]);
+    
+    if(BH == ThisParticle->ParticleClass)
+      continue;
+    /* The first form of feedback is accretion luminosity 
+     * This is given by L = G M_acc M_* / R_*
+     */
+    /* StarMass in units of Msolar */
+    float StarMass = ThisParticle->Mass * MassConversion / SolarMass;
+    /* Accretion Rate in units of Msolar/s */
+    float AccretionRate = ThisParticle->AccretionRate[ThisParticle->TimeIndex]*MassUnits/(SolarMass*TimeUnits);
+    if(AccretionRate*3.154e7 < 1e-30) /* Takes care of any negative accretion rates */
+      AccretionRate = 1e-30/3.154e7;
+    /* Star Radius in units of Rsolar */
+    float StarRadius = GetStellarRadius(StarMass, AccretionRate);
+    float LThisTimestep=0.0, sn_nrg_thistimestep = 0.0;
+    if((POPIII == ThisParticle->ParticleClass) || (SMS == ThisParticle->ParticleClass)) {
+      /* Calculate SpecificL [ergs s^-1 g^-1] */
+      float SpecificL = GravConst * AccretionRate * SolarMass / (StarRadius * SolarRadius);
+      SpecificL = SpecificL/LumConvert;  /* Convert (specific) Luminosity to code units */
+#if SSDEBUG
+      printf("%s: dx = %e\t MassConversion = %e\n", __FUNCTION__, dx, MassConversion);
+      printf("%s: AccretionRate = %e Msolar/yr (Code = %e)\n", __FUNCTION__,
+	     AccretionRate*3.154e7, ThisParticle->AccretionRate[ThisParticle->TimeIndex]);
+      printf("%s: Mass = %e Msolar\n", __FUNCTION__, StarMass);
+      printf("%s: Radius = %e Rsolar\n", __FUNCTION__, StarRadius);
+      printf("%s: SpecificL Total = %e Lsolar\n", __FUNCTION__, SpecificL*LumConvert * StarMass * SolarMass/SolarLuminosity);
+#endif
+    /* Now calculate the energy over the timestep */
+      LThisTimestep = SpecificL*dt/float(StarFeedbackDistTotalCells);
+#if SSDEBUG
+      printf("%s: StarFeedbackDistTotalCells = %f\n", __FUNCTION__,
+	     float(StarFeedbackDistTotalCells));
+      printf("%s: dt = %e yrs\n", __FUNCTION__,
+	     dt*TimeUnits/3.154e7);
+      printf("%s: L Total = %e Lsolar [ergs]\n", __FUNCTION__,
+	     LThisTimestep*(StarMass * SolarMass));
+#endif   
+      
+      float Age = Time - ThisParticle->BirthTime;
+#if SSDEBUG      
+      printf("%s: Star Age = %e yrs\n", __FUNCTION__, Age*TimeUnits/3.154e7);
+      printf("%s: Radiation Lifetime =  %e yrs\n", __FUNCTION__, ThisParticle->RadiationLifetime*TimeUnits/3.154e7);
+      printf("%s: Particle Class = %d\t POPIII = %d\n", __FUNCTION__, ThisParticle->ParticleClass, POPIII);
+#endif
+      /* Check for star death and transition to BH */
+      if(ThisParticle->RadiationLifetime < Age) {/* Star needs to go supernovae and change type */
+#ifdef SNEFEEDBACK
+	/* Based on the stellar mass calculate the SN energy and ejecta mass */
+	if(POPIII == ThisParticle->ParticleClass) {
+	  float he_core = (13.0/24.0) * (StarMass - 20.0);
+	  sn_nrg_thistimestep = (5.0 + 1.304 * (he_core - 64.0)) * 1e51; //ergs
```

Comment:
Is the supernova energy always 1e51? Or is this a parameter that can be specified by the user?

Reply:
This is a fit from Heger & Woosley (2002), and I believe it shouldn't be user specified.

Reply:
I took this fit from other PopIII models in Enzo. I couldn't find a paper that references this fit though. It would be good to include it though.

Reply:
Put in a comment